### PR TITLE
Fix bug with storage of "nodedef" metadata on RtNodeGraph

### PR DESCRIPTION
**Pull request by niklasharrysson**

### CODE REVIEW

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ namespace
     static const RtTokenSet inputMetadata       = { RtToken("name"), RtToken("type"), RtToken("value"), RtToken("nodename"), RtToken("output"), RtToken("channels"), 
                                                     RtToken("nodegraph"), RtToken("interfacename") };
     static const RtTokenSet nodeMetadata        = { RtToken("name"), RtToken("type"), RtToken("node") };
-    static const RtTokenSet nodegraphMetadata   = { RtToken("name") };
+    static const RtTokenSet nodegraphMetadata   = { RtToken("name"), RtToken("nodedef") };
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtFileIo.cpp:40]: Declare that "nodedef" metadata is handled explicitly.

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 2]
```diff
@@ -375,6 +375,7 @@ namespace
         if (srcNodeDef)
         {
             createInterface(srcNodeDef, schema);
+            schema.setDefinition(RtToken(srcNodeDef->getName()));
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtFileIo.cpp:378]: Handle setting nodedef explicitly.

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 3]
```diff
@@ -908,6 +909,12 @@ namespace
 
         RtNodeGraph nodegraph(src->hnd());
 
+        const RtToken& nodedef = nodegraph.getDefinition();
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtFileIo.cpp:912]: Handle writing out nodedef explicitly.

File: source/MaterialXTest/MaterialXRuntime/Runtime.cpp [hunk 4]
```diff
@@ -3,8 +3,6 @@
 // All rights reserved.  See LICENSE.txt for license.
 //
 
-#ifdef MATERIALX_BUILD_RUNTIME
```
Comment (niklasharrysson) [inline source/MaterialXTest/MaterialXRuntime/Runtime.cpp:6]: Should not be used, the file is not added by cmake if the flag is not set.

**Review by feldstj — APPROVED** [review]

**Pull request merged**